src/components/montecarlo/histogram-panel.tsx

```diff
@@ -32,19 +44,99 @@ export class HistogramPanel extends BaseComponent<IProps, IState>{
   constructor(props: IProps) {
     super(props);
     const initialState: IState = {
-      bars: false,
+      showBars: false,
     };
     this.state = initialState;
   }
 
   public render() {
     const {width, height} = this.props;
+    const histogramChart = this.stores.chartsStore.charts.find(chart => chart.type === "histogram");
 
     return (
       <Panel height={height} width={width}>
-        MONTE CARLO
+        { histogramChart
+          ? this.renderHistogram(histogramChart)
+          : <button onClick={this.addHistogram}>Add histogram</button>
+        }
       </Panel>
     );
   }
 
+  private renderHistogram = (chart: ChartType) => {
+    const { width, height } = this.props;
+    const { data } = chart;
+
+    const dataArray: any[] = data.map(d => d[0]);
+    const threshold = kTephaThreshold;
+    let above = 0;
+    dataArray.forEach(d => {
+      if (d > threshold) {
+        above++;
+      }
+    });
+    const below = dataArray.length - above;
+    const abovePercent = Math.round(above / dataArray.length * 100);
+    const belowPercent = 100 - abovePercent;
+
+    return (
+      <HorizontalContainer>
+        <SvgD3HistogramChart
+          width={width - 200}
+          height={height - 10}
+          chart={chart}
+          chartMin={kTephaMin}
+          chartMax={kTephaMax}
+          showBars={this.state.showBars}
+        />
+        <VerticalContainer>
+          <PanelStat>{`Threshold = ${threshold}mm`}</PanelStat>
+          <PanelStat>{`Count below threshold: ${below} (${belowPercent}%)`}</PanelStat>
+          <PanelStat>{`Count above threshold: ${above} (${abovePercent}%)`}</PanelStat>
+          <PanelStat>{`Risk: ${this.calculateRisk(abovePercent)}`}</PanelStat>
+          <button onClick={this.changeDisplayType}>Toggle display</button>
+        </VerticalContainer>
+      </HorizontalContainer>
+    );
+  }
+
+  private calculateRisk = (percent: number) => {
+    if (percent > 80) {
+      return "High";
+    } else if (percent > 31) {
+      return "Medium";
+    } else {
+      return "Low";
+    }
+  }
+
+  // TODO: for testing only, remove once we decide on plot style
+  private changeDisplayType = () => {
+    this.setState(prevState => ({
+      showBars: !prevState.showBars
+    }));
+  }
+
+  // TODO: for testing only, remove when we have real data
+  private addHistogram = () => {
+      const numPoints = 300;
+      const title = "Chart " + (this.stores.chartsStore.charts.length + 1);
+      const data: number[][] = [];
+      for (let i = 0; i < numPoints; i++) {
+        data.push([Math.floor(this.randomG(3) * (kTephaMax + 1))]);
+      }
```

Comment:
I don't quite understand the data structure. Why is it a 2d array single-length arrays? Why not use a 1d array?

Reply:
I was trying to be consistent with how the existing chart model structured data:
`types.array(types.array(types.union(types.number, types.Date)))`

Reply:
But it's probably better to expand the `data` type definition in the `charts-stores.ts`

Reply:
So we could do something like this:
`data: types.array(types.union(types.array(types.union(types.number, types.Date)),types.number))` in the `Chart` model in `charts-stores.ts`.  The data can be:
- an array of numbers
- an array of arrays of numbers or dates
This does break some other code so I believe we'll have to do some extra checking to ensure we have a 2d array in some cases.  Something like `Array.isArray(self.data[x])`.

Reply:
I added a commit that allows that data to also be stored as a simple array of numbers.  
Let me know what you think:
https://github.com/concord-consortium/geocode/pull/124/commits/1fdb0a842d6cdece7b071efb21a0d328ec14b67b